### PR TITLE
Added the LaborApproval service.

### PR DESCRIPTION
**Pull request by jordanyaker**
Added the service for handling the approval requests.

### CODE REVIEW

File: lib/laborApproval.js [hunk 1]
```diff
@@ -0,0 +1,107 @@
+const ServiceBase = require('./serviceBase');
+const SimpleSchema = require('simpl-schema').default;
+const moment = require('moment');
+
+const SERVICE = 'Erp.Bo.LaborApprovalSvc';
+
+// Build a [where, whereDtl] condition based on some pre-defined fields
+const buildWhere = criteria => {
```
Comment (nicocrm) [inline lib/laborApproval.js]: Can we share this one with the one in Labor.js, or is there a subtle difference between them?

File: lib/laborApproval.js [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+    where.push(`OprSeq = '${OprSeq}'`);
+  }
+  if (IndirectCode) {
+    where.push(`IndirectCode = '${IndirectCode}'`);
+  }
+
+  return where.join(' and ');
+};
+
+// labor service - this combines laborhed and labordtl records, so there is a good bit of custom logic
```
Comment (nicocrm) [inline lib/laborApproval.js]: The comment must have been pasted from labor.js

File: lib/laborApproval.js [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+      dataset.LaborDtl.forEach(ld => {
+        const payrollDate = moment(ld.PayrollDate).format('YYYY-MM-DD');
+
+        if (datesToApprove.includes(payrollDate)) {
+          Object.assign(ld, {
+            ApprovalProjectDesc: ld.ProjectDescription,
+            ApprovalProjectID: ld.ProjectID,
+            ApprvrHasOpenTask: true,
+            ApprovedBy: approvedByName,
+            ApprovedDate: moment().format('YYYY-MM-DDT00:00:00'),
```
Comment (nicocrm) [inline lib/laborApproval.js]: do we need to pass this?  It seems like it was set automatically when I tried the call in postman

Comment (jordanyaker) [reply lib/laborApproval.js]: Let me actually give it a try and see what data it _really_ needs.

**Pull request merged**